bitwarden_license/bit-web/src/app/sm/secrets/dialog/secret-delete.component.ts

```diff
@@ -0,0 +1,56 @@
+import { DialogRef, DIALOG_DATA } from "@angular/cdk/dialog";
+import { Component, Inject } from "@angular/core";
+
+import { ValidationService } from "@bitwarden/angular/services/validation.service";
+import { I18nService } from "@bitwarden/common/abstractions/i18n.service";
+import { LogService } from "@bitwarden/common/abstractions/log.service";
+import { PlatformUtilsService } from "@bitwarden/common/abstractions/platformUtils.service";
+
+import { SecretService } from "../secret.service";
+
+export interface SecretDeleteOperation {
+  secretIds: string[];
+}
+
+@Component({
+  selector: "sm-secret-delete-dialog",
+  templateUrl: "./secret-delete.component.html",
+})
+export class SecretDeleteDialogComponent {
+  constructor(
+    public dialogRef: DialogRef,
+    private secretService: SecretService,
+    private i18nService: I18nService,
+    private platformUtilsService: PlatformUtilsService,
+    private logService: LogService,
+    private validationService: ValidationService,
+    @Inject(DIALOG_DATA) public data: SecretDeleteOperation
+  ) {}
+
+  get title() {
+    return this.data.secretIds.length === 1 ? "deleteSecret" : "deleteSecrets";
+  }
+
+  async delete() {
+    try {
+      await this.secretService.delete(this.data.secretIds);
+      this.dialogRef.close();
+      if (this.data.secretIds.length === 1) {
+        this.platformUtilsService.showToast(
+          "success",
+          null,
+          this.i18nService.t("softDeleteSuccessToast")
+        );
+      } else {
+        this.platformUtilsService.showToast(
+          "success",
+          null,
+          this.i18nService.t("softDeletesSuccessToast")
+        );
+      }
```

Comment:
I think this looks a bit cleaner.

```suggestion
      const message = this.data.secretIds.length === 1 ? "softDeleteSuccessToast" : "softDeletesSuccessToast";
      this.platformUtilsService.showToast("success", null, this.i18nService.t(message));
```

Reply:
Resolved in https://github.com/bitwarden/clients/pull/3549/commits/4cc5f9cab763768911ece8f63b982565b638b67f